pkg/util/ipsec/ipsec_telemetry.go

```diff
@@ -0,0 +1,50 @@
+package ipsec
+
+import (
+	"sync"
+
+	operv1 "github.com/openshift/api/operator/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+var (
+	registerMetrics sync.Once
+	ipsecStateGauge *metrics.GaugeVec
+
+	recordedMode, recordedFlavor string = "", ""
+)
+
+// RegisterIpsecMetrics in the global registry
+func RegisterIpsecMetrics() {
+	registerMetrics.Do(func() {
+		ipsecStateGauge = metrics.NewGaugeVec(&metrics.GaugeOpts{
+			Name: "cluster_network_ipsec_state",
+			Help: "A metric with a constant '1' value labeled by the latest ipsecMode of the cluster",
+		}, []string{"mode", "apiFlavor"})
```

Comment:
```suggestion
		}, []string{"mode", "api_flavor"})
```
Prometheus has a strong preference for snake_case oder camelCase.